From 1b74b520b89217b7261bb11f20232aab01565899 Mon Sep 17 00:00:00 2001
From: Brennan Lamoreaux <blamoreaux@vmware.com>
Date: Fri, 7 Jul 2023 16:42:44 +0000
Subject: [PATCH 2/2] falcoctl: build locally

falcoctl is consumed as a precompiled Go binary, potentially built
with outdated Go version. Compile it locally with the latest Go version.
---
 cmake/modules/falcoctl.cmake | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/cmake/modules/falcoctl.cmake b/cmake/modules/falcoctl.cmake
index 52c2e786..8e83128a 100644
--- a/cmake/modules/falcoctl.cmake
+++ b/cmake/modules/falcoctl.cmake
@@ -17,20 +17,16 @@ string(TOLOWER ${CMAKE_HOST_SYSTEM_NAME} FALCOCTL_SYSTEM_NAME)
 
 set(FALCOCTL_VERSION "0.5.0")
 
-if(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "x86_64")
-    set(FALCOCTL_SYSTEM_PROC_GO "amd64")
-    set(FALCOCTL_HASH "ba82ee14ee72fe5737f1b5601e403d8a9422dfe2c467d1754eb488001eeea5f1")
-else() # aarch64
-    set(FALCOCTL_SYSTEM_PROC_GO "arm64")
-    set(FALCOCTL_HASH "be145ece641d439011cc4a512d0fd2dac5974cab7399f9a7cd43f08eb43dd446")
-endif()
+set(FALCOCTL_HASH "bcb8c46e7e0e22e8958135daf35bc78380877eea4027bae9f459ad1d6c2cf1ad")
 
 ExternalProject_Add(
         falcoctl
-        URL "https://github.com/falcosecurity/falcoctl/releases/download/v${FALCOCTL_VERSION}/falcoctl_${FALCOCTL_VERSION}_${FALCOCTL_SYSTEM_NAME}_${FALCOCTL_SYSTEM_PROC_GO}.tar.gz"
+        URL "https://github.com/falcosecurity/falcoctl/archive/refs/tags/v${FALCOCTL_VERSION}.tar.gz"
         URL_HASH "SHA256=${FALCOCTL_HASH}"
         CONFIGURE_COMMAND ""
-        BUILD_COMMAND ""
+        SOURCE_DIR "falcoctl-${FALCOCTL_VERSION}"
+        BUILD_IN_SOURCE "true"
+        BUILD_COMMAND "make"
         INSTALL_COMMAND "")
 
-install(PROGRAMS "${PROJECT_BINARY_DIR}/falcoctl-prefix/src/falcoctl/falcoctl" DESTINATION "${FALCO_BIN_DIR}" COMPONENT "${FALCO_COMPONENT_NAME}")
+install(PROGRAMS "${PROJECT_BINARY_DIR}/falcoctl-${FALCOCTL_VERSION}/falcoctl" DESTINATION "${FALCO_BIN_DIR}" COMPONENT "${FALCO_COMPONENT_NAME}")
-- 
2.39.0

